File: frontend/src/administrator/team/service/teamService.ts
import Page from "@/@core/utils/page";
import { http } from "@/plugins/axios/axios";
import Team from "../model/team";
import TeamResponse from "../model/teamResponse";

export default class TeamService {
    private _endpoint = "/api/v1/teams";

    async save(team: Team): Promise<Team> {
        if (team.id === null) {
            return await this.create(team);
        }

        return this.update(team);
    }

    private async create(team: Team): Promise<Team> {
        const response = await http.post(this._endpoint, team);

        return response.data;
    }

    private async update(team: Team): Promise<Team> {
        const response = await http.put(`${this._endpoint}`, team);

        return response.data;
    }

    async findById(id: number): Promise<Team> {
        const response = await http.get(`${this._endpoint}/${id}`);

        return response.data;
    }

    async findByFilter(): Promise<Page<TeamResponse>> {
        const response = await http.get<Page<TeamResponse>>(
            `${this._endpoint}/filter`,
        );

        return response.data;
    }
}

File: backend/team/api/src/main/java/br/com/systec/taskflow/team/api/service/TeamService.java
package br.com.systec.taskflow.team.api.service;

import br.com.systec.taskflow.commons.exceptions.BaseException;
import br.com.systec.taskflow.team.api.filter.TeamFilter;
import br.com.systec.taskflow.team.api.vo.TeamVO;
import org.springframework.data.domain.Page;

public interface TeamService {

    TeamVO create(TeamVO teamVO) throws BaseException;

    TeamVO update(TeamVO teamVO) throws BaseException;

    TeamVO findById(Long id) throws BaseException;

    void delete(Long id) throws BaseException;

    void addMember(Long teamId, Long employeeId) throws BaseException;

    void removeMember(Long teamId, Long memberId) throws BaseException;

    Page<TeamVO> findByFilter(TeamFilter filter) throws BaseException;
}
